src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```diff
@@ -0,0 +1,216 @@
+package net.sf.jabref.logic.importer.fetcher;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.sf.jabref.logic.help.HelpFile;
+import net.sf.jabref.logic.importer.FetcherException;
+import net.sf.jabref.logic.importer.IdBasedParserFetcher;
+import net.sf.jabref.logic.importer.ImportFormatPreferences;
+import net.sf.jabref.logic.importer.Parser;
+import net.sf.jabref.logic.importer.ParserResult;
+import net.sf.jabref.logic.importer.SearchBasedFetcher;
+import net.sf.jabref.logic.importer.fileformat.MedlineImporter;
+import net.sf.jabref.model.entry.BibEntry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * Fetch or search from Pubmed http://www.ncbi.nlm.nih.gov/sites/entrez/
+ */
+public class MedlineFetcher implements IdBasedParserFetcher, SearchBasedFetcher {
+
+    private static final Log LOGGER = LogFactory.getLog(MedlineFetcher.class);
+
+    private static final Pattern ID_PATTERN = Pattern.compile("<Id>(\\d+)</Id>");
+    private static final Pattern COUNT_PATTERN = Pattern.compile("<Count>(\\d+)<\\/Count>");
+    private static final Pattern RET_MAX_PATTERN = Pattern.compile("<RetMax>(\\d+)<\\/RetMax>");
+    private static final Pattern RET_START_PATTERN = Pattern.compile("<RetStart>(\\d+)<\\/RetStart>");
+
+    private ImportFormatPreferences importFormatPreferences;
+
+    private int count;
+    private int retmax;
+    private int retstart;
+    private String ids = "";
+
+    public MedlineFetcher(ImportFormatPreferences importFormatPreferences) {
+        this.importFormatPreferences = importFormatPreferences;
+    }
+
+    private void addID(String id) {
+        if (ids.isEmpty()) {
+            ids = id;
+        } else {
+            ids += "," + id;
+        }
+    }
+
+    /**
+     * Removes all comma's in a given String
+     *
+     * @param query input to remove comma's
+     * @return input without comma's
+     */
+    private static String replaceCommaWithAND(String query) {
+        return query.replaceAll(", ", " AND ").replaceAll(",", " AND ");
+    }
+
+    /**
+     * When using 'esearch.fcgi?db=<database>&term=<query>' we will get a list of ID's matching the query.
+     * Input: Any text query (&term)
+     * Output: List of UIDs matching the query
+     *
+     * @see <a href="https://www.ncbi.nlm.nih.gov/books/NBK25500/">www.ncbi.nlm.nih.gov/books/NBK25500/</a>
+     */
+    private void getPubMedIdsFromQuery(String query, int start, int pacing) throws FetcherException {
```

Comment:
Please try to return here a list of IDs instead of calling `addId` (see my previous review)